Neon fates

adrifoster opened the pull request:
### Description of changes

Small updates to facilitate creation and modification of FATES-usable (i.e. 16-PFT) NEON surface data files.

### Specific notes

I updated `neon_surf_wrapper.py` and `modify_singlept_site_neon.py` to include a `--fates` argument that will create and/or modify the 16-PFT versions of the surface datasets. Right now FATES can't run with the 78-PFT versions and we aren't sure when this will get fixed so this is a somewhat temporary change.

The `--fates` argument just modifies the file names in the [find_surffile](https://github.com/adrifoster/CTSM/blob/f1137fd74348359ff70c7952f550b1812d3e6217/tools/site_and_regional/modify_singlept_site_neon.py#L273) function.

I also created a set of new user mods in cime_config/usermods_dirs/NEON/FATES. As per suggestion from These are the exact same user_mods files as in the big-leaf version but with the surface data file name (in the `user_nl_clm` file) updated to reference the new files I created with these scripts (in /glade/p/cesmdata/inputdata/lnd/clm2/surfdata_map/NEON/FATES). 

The lines about lightning files were also deleted as these are not allowed in a FATES run (@ekluzek I'm not sure if this is also something we need to update in the FATES code to run NEON?)

Finally the `hist_fincl2` names were updated to FATES-specific ones (again something we need to think about @ekluzek).

_A big issue here is the replication of all the directories for the FATES user mods - What do we think?_
The main problem is that when we reference a NEON site user_mods we reference the folder (i.e. /user_mods/NEON/ABBY) and then *that* user_mods references the `../defaults` in the `include_user_mods` file. So for the FATES-specific `defaults` files, we need everything to be in its own encompassing folder. 

_Can we somehow modify the way `include_user_mods` is read to allow for variables?_ My idea would be for the `include_user_mods` to say something like:
```
../defaults_${version}
```
And then in `shell_commands` we would set up `version=fates` or `version=BL` or something. We could then have 2 *defaults* folders (*defaults_fates*, *defaults_BL*). @billsacks what do you think? This would I _think_ be the least duplicative...

Contributors other than yourself, if any: @ekluzek, @billsacks, @TeaganKing

CTSM Issues Fixed: Working towards #1609

Are answers expected to change? No

Any User Interface Changes? No

Testing performed, if any:

Tested the ABBY NEON site and seemed to run fine, can test all NEON sites if needed.

Will run `aux_clm` and `fates` test suites - should probably create a test for FATES-NEON?

adrifoster commented:
In talking with @jkshuman it might be good to also push this through before the winter meeting so that we can have FATES/NEON at least functional, if not the optimal method of setting up the cases. We can then create a new PR to get rid of all the hacky user-mods stuff.

jkshuman commented:
Agree with @adrifoster that being able to present this NEON FATES functionality at the winter meeting would be useful. Then further optimization can happen behind the scenes, since the functionality will not change.

wwieder commented:
+1, it would be great to illustrate these capabilities by the WG meeting!  Is it critical to get a PR merged to main for this to happen? I'm trying to be aware of demands on everyone's time, especially @ekluzek .  

Meanwhile @teaganking is working on #1904.  I wonder if it's worth having a focused discussion on the best way to implement these more diverse functionalities into the NEON workflow?  Would our Thursday CLM meeting be too soon to start discussing this?

adrifoster commented:
> +1, it would be great to illustrate these capabilities by the WG meeting! Is it critical to get a PR merged to main for this to happen? I'm trying to be aware of demands on everyone's time, especially @ekluzek .

If we want other people to be able to run FATES NEON then yes, otherwise we would have to reference my code or tell them what files to modify - which we can do now

wwieder commented:
sorry, maybe I wasn't clear with my question.  Yes, we want this on main, but does it need to come to main before the WG meeting?  

On the topic of _"hacky user-mods stuff"_ is there a simple way of using run_neon to handle some of this? How are you actually defining this as a FATES case? It seems like the only new stuff in usermods is related to default/user_nl_clm, is there some way to use the compset we're using to key off what defaults to add? (maybe you and Erik have already discussed this?)

adrifoster commented:
> sorry, maybe I wasn't clear with my question. Yes, we want this on main, but does it need to come to main before the WG meeting?

I guess we would want it on main before the WG meeting if we want to showcase it and then have folks be able to go do it themselves, which I think would be good.

> On the topic of "hacky user-mods stuff" is there a simple way of using run_neon to handle some of this? How are you actually defining this as a FATES case? It seems like the only new stuff in usermods is related to default/user_nl_clm, is there some way to use the compset we're using to key off what defaults to add? (maybe you and Erik have already discussed this

The main issue is just that we have multiple folders of user_mods, which works fine but just requires us to update things in two places if updates need to occur (for example with PR #1933). From the user's perspective it will work (the way the PR is right now) the same as a big-leaf method. Just set user_mods to the path to the folders I created. I haven't updated `run_neon` but that wouldn't be too difficult and we can do that eventually. Right now to run a FATES NEON case it would be similar to creating any other CTSM job but using a FATES compset and referencing the user_mods for the FATES NEON folder:

```
NEON_SITE=ABBY

./create_newcase --case ${NEON_SITE}_test --res CLM_USRDAT --compset 2000_DATM%1PT_CLM50%FATES_SICE_SOCN_SROF_SGLC_SWAV --run-unsupported --user-mods-dir /CTSM/cime_config/usermods_dirs/NEON/FATES/${NEON_SITE}
```
We can add this functionality to `run_neon` as well but I think @ekluzek was thinking this could happen at a later date. Right now the folks who are asking about FATES and NEON are regular FATES users so would not use `run_neon`. @jkshuman correct?

wwieder commented:
OK, maybe we can combine this with #1933, which would require duplicate changes to usermods_dir/NEON/ONAQ and also updating the 16 PFT surface datasets for the site. With improvements to the overall infrastructure to be done at a later date.
 
We can discuss at the SE meeting on Thurs.

jkshuman commented:
@adrifoster I agree that having it on main before meeting is the best and cleanest way to showcase this. It is on main and not in a user directory. I also agree that FATES users will not be using run_neon to start. run_neon would be a potential future development. We can discuss further on Thursday.

billsacks commented:
> Can we somehow modify the way include_user_mods is read to allow for variables?

Interesting. I like the general idea. I can't quite picture how your specific suggestion would work (not necessarily an issue with your suggestion - I may just need you to explain it to me in more detail). One thing to note is that, as currently implemented, the include directories are built up recursively before the shell commands are parsed - see https://github.com/ESMCI/cime/blob/master/CIME/user_mod_support.py (called from the apply_user_mods function in https://github.com/ESMCI/cime/blob/master/CIME/case/case.py). One way I could see this working is to introduce a new env_case xml variable like "include_user_mods_var"; this could be set in config_component.xml based on the compset, and I think apply_user_mods would have access to this.

But I wonder if it's also worth considering some alternative ways to get at this. Specifically, it's now possible to specify multiple `--user-mods-dirs` on the create_newcase command line (and also to specify a user_mods_dir that is set automatically via the compset, though I'm not sure if that would be useful here). Would it make sense, rather than using include_user_mods, to specify two user-mods-dirs up-front – one for the specific NEON site and then the appropriate "default" based on whether this is a FATES or non-FATES case? I haven't thought this through carefully, so this may not work, but I wanted to put this possibility out there so you could judge whether it could work.

ekluzek commented on cime_config/usermods_dirs/NEON/FATES/defaults/user_nl_clm:
```diff
@@ -0,0 +1,28 @@
+!----------------------------------------------------------------------------------
```
In above I suggest that this would look like an include_user_mods file that looks like

../../_includes/defaults

Then in cime_config/usermods_dirs/NEON/FATES/defaults you'd just have the FATES specific changes which I think is just the history variable list and the surface dataset filename

In the big-leaf defaults it would similarly have a include_user_mods directory that just points to the baseclass version and then includes the regular surface dataset and the history fields list for big-leaf.

adrifoster replied:
Leave for now but know that we are going to update this in the future.

ekluzek reviewed (changes requested):
OK, I love that you are working on this and it's nice to see it here. I made a bunch of specific suggestions on changes. I think you should do most of them. But, we can discuss as much of it as you like.

I propose that we have the sites specific files be divided into three lists of site specific versions. The first would be the baseclass that's used by both, the second is the big-leaf version that just points to the baseclass file and the big-leaf default. The third is the FATES set that points to the baseclass version and the FATES default. This unfortunately duplicates the site list directories -- but won't require double maintenance.  It will require that new sites go to all three directories -- but that is rare so it's acceptable.

I also suggest that we remove the word FATES for some of these changes, specifically from the ones that aren't fundamental to FATES and that will go away soon (hopefully).

Then I have some specific comments and questions throughout.

By the way welcome back! Nice to see you already contributing.

ekluzek commented:
@jkshuman and @adrifoster we talk about run_neon.py and propose working on it later as it sounds like it might be a big effort. Looking at run_neon.py, I don't think it will be *that* hard. You mainly add an option for --fates, and then a few things need to be different if it's flagged.

The one thing I'm not sure about is the spinup procedure for FATES. For big-leaf you do, AD, post-AD, and then transient. For FATES you can't do transient, but I think you'd still do AD and post-AD right? Is there anything else that would be different?

I'll add this to tomorrows agenda for more discussion.

adrifoster commented on cime_config/usermods_dirs/NEON/FATES/STER/shell_commands:
```diff
@@ -0,0 +1,7 @@
+# Setup to run with prognostic crops for this site
+# If you want to explicitly run in SP mode or add other
+# # options you'll need to add that after this...
+./xmlchange CLM_BLDNML_OPTS="--bgc bgc --crop"
```
Hmmm yeah I'm not sure what to do about these crop sites...

ekluzek replied:
The three options that I see would be to:
- Delete the crop site for FATES, so it wouldn't be an option
- Run it without crop for FATES, but maybe only allow a single grassland PFT? So the configuration would be a little different than other sites.
- Keep the directory, but set it up so that it fails with a warning that this is a crop site and you can't run FATES that way

I think it would be OK to delete the crop sites for FATES since you probably do want to run the full prognostic crop model for these sites, and you would do that from the big-leaf version. As such possibly that last option where it fails with an error is the best.

adrifoster replied:
> I think it would be OK to delete the crop sites for FATES since you probably do want to run the full prognostic crop model for these sites, and you would do that from the big-leaf version. As such possibly that last option where it fails with an error is the best

Okay that makes sense to me. How should we fail the run?

ekluzek replied:
We can add an exit statement to the bash shell_commands script.

adrifoster commented on tools/site_and_regional/neon_surf_wrapper.py:
```diff
@@ -52,6 +52,12 @@ def get_parser():
                 default=False)
 
+    parser.add_argument('-f', '--fates',
+                        help='Create and/or modify surface data file for FATES. ',
```
Sounds good to me.

adrifoster replied:
Actually here we may want to leave the FATES signifier since we (in my opinion) do things FATES-specific like not choosing one dominant PFT. I can change to a `--16pft` flag for the `modify_singlept_site_neon` script though?

ekluzek replied:
Ahhh, good point. Hmmm. Although I still think the standard thing would be to do the dominate PFT. Full FATES will ignore that so that's sufficient for that and cases you want to replicate the BGC setup. It's the "in between" modes that someone might want to have multiple PFT's where it matters here.

ekluzek replied:
Should we have a quick call to discuss?

adrifoster replied:
> Should we have a quick call to discuss?

Yes lets try to include @jkshuman as well?

ekluzek replied:
I know Jackie is busy this week. She's trying to get a manuscript done. We can always ask though.

adrifoster replied:
@ekluzek just met briefly. The issue is that the 1 degree surface files are "wrong" with respect to what the NEON data says is actually present at the site, but right now we only have data on 1 dominant PFT per site. So for now we **are** going to just go with 1 dominant PFT for the surface files. Remember that in full FATES mode you can still have multiple PFTs, based on your FATES parameter file. 

In the future we would like to update the subset surface data file to have the PFT mixture match what is observed at the NEON site (using inventory data, etc.). I will make an issue about this.

jkshuman replied:
@ekluzek where is the dominant PFT information coming from if the surface data files are wrong? Does it make sense to wait until https://github.com/ESCOMP/CTSM/issues/1934 is implemented so that we can run mixed PFT FATES simulations?

ekluzek replied:
Here's what @adrifoster and I thought at our discussion yesterday. First the dominant PFT information IS correct. It is coming from NEON, and we think it's all correct. What we currently don't have are the inventory files, nor a list of the PFT's that are at the site. So implementing #1934 will take both software infrastructure work, as well as science research to express inventory data into % cover.

The part that would be wrong is to use the 1-degree mix of PFT's from the global 1-degree dataset and say that represents the site. We could turn off dominant PFT and use the 1-degree mix, and for some sites that might be OK. But, I'd expect it to be wrong and too extensive for most sites. We could do a check and see what the list is for the sites when run this way though -- maybe I'm wrong and the list is reasonable if not ideal. It's just when I look out at the world a local small site normally has a really restricted set of PFT's, while a large 1-degree grid-cell is going to have tons of variety.

Since, it will take time to do #1934 we figured starting with the dominant PFT makes sense. For full FATES you'll still get whatever mix of PFT's you want to have, so it's fine for full FATES. It's also fine for FATES-SP mode where you just want to compare to CTSM-BGC fro the simplest possible case. What it is missing is the complexity cascade of going from FATES-SP, to fixed biogeography, to NOCOMP, to full FATES where you really want to have the mix of PFT's. But, in terms of getting something in place now it's a nice baby step. It also means that people can readily run FATES for NEON sites and at least get their feet wet. You are right to take a full dive, we need #1934, but as it's going to be complex enough and take enough time -- we figure doing a baby step would be the right thing to do.

We could do a group discussion if that helps? Maybe next week?

jkshuman replied:
@ekluzek I ask about this b/c at the SE meeting yesterday we had all agreed that the implementation was to run FATES with mixed PFTs as this allows for testing across modes. The mix of PFTs is an important piece for calibration and validation. The desire to skip the surface dataset and use this NEON specific dataset with only the dominant PFT was not introduced yesterday as a critical limiting factor for providing FATES-NEON function. This is an important point. There will be differences in calibrating to 1 PFT vs a mix of PFTs, but this PR addresses only function and not calibration. 

There are a number of ways to determine dominant PFT in a forest stand, and many of these NEON sites do in reality have multiple species and potentially multiple PFTs present in the inventory data. NEON probably has a method to provide this dominant PFT, so it would be good to have that information in this NEON file as a comment. It seems the suggestion is to now use the NEON specific dataset where the only data provided is the dominant 1 PFT. 

Then with https://github.com/ESCOMP/CTSM/issues/1934 the PFT mix will be adjusted based on reality from the inventory.

wwieder replied:
Currently when I generate surface data for NEON sites I use this [wrapper](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_surf_wrapper.py) which get's dominant PFT (based on expert opinion) from this ([.csv file](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_sites_dompft.csv)). 

We could define additional (common) pfts in the this file to be read in for FATES surface data, but again this seems like it would take a bit longer to do well enough to be defensible. FWIW, I like the baby-step approach we're taking, with a more robust estimate of veg cover coming in with #1934.  Based on what Adrianna showed yesterday it seems like this isn't far from being ready to go NEON-wide?

ekluzek replied:
Sorry @jkshuman I really do not mean to go against our agreements from the meeting. But, a challenge we have is that even though we were both in the same meeting we all have different expertise and so pick up on different things. And we might all might miss something stated because of loss of focus. Since, I don't live in FATES I didn't pick up on the need to ensure there are multiple PFT's. What I am trying to do is to make sure we aren't doing something that struck me as wrong. I did honestly think in the meeting that we were talking about having the datasets use the dominant PFT the same as the CTSM-BGC datasets.

What I hear you saying is that you'd like to use the 1-degree gridcell PFT's for this first go around. And then later we will get the corrected PFT distribution from the NEON inventory data. Is that right? So the first baby-step is to use the 1-degree PFT's, and then with #1934 we'll get the correct ones.

It struck me as the most correct to use the dominate PFT as that's from the actual site. But, you and @adrifoster are the experts in FATES here, so it should be done in the way you think is most correct.  I think it is important to realize it is wrong either way, for this first baby step, but yes you and @adrifoster should decide which way you think it should be for this first step. The ones that are the experts in the question at hand should be the ones that decide.

So sorry for derailing the train -- I really didn't mean to do that. :-(

adrifoster replied:
Thanks @ekluzek 

I wanted to provide an example of how this looks in these forests. From an ecological perspective, it is important to include more than just 1 PFT at many of these sites. In many locations even at the individual-plot level (i.e. a few hundred meters squared) we can see a large number of species of differing PFTs. For example, looking at the inventory plots for HARV, which surround the flux tower and are intended to be representative of the forest, we can see a huge mix of species (upwards of 50 species), and even reducing to just the genus we still have quite a mix (~35 genera):

![by_genus](https://user-images.githubusercontent.com/13225250/212429040-27ea1608-387c-48ec-80dd-df68791cfaa7.png)

If we further reduce that to PFTs, we mostly get a mix of deciduous broadleaf and needleleaf evergreen trees (about 60% deciduous, and 40% evergreen). From an ecological perspective, I don't think it is appropriate to just say this site is "deciduous broadleaf", even though that is what the [csv file ](https://github.com/ESCOMP/CTSM/blob/master/tools/site_and_regional/neon_sites_dompft.csv) classifies it as.

![by_pft](https://user-images.githubusercontent.com/13225250/212429275-fb8587a1-c0ef-411c-bb2b-4d317b4c35e4.png)

Here are averages across the inventory plots for ABBY, BONA, and HARV:

![average_pft](https://user-images.githubusercontent.com/13225250/212429720-0aba60c3-0cab-4e09-8c79-733938b0589a.png)

As you can see, while ABBY is pretty dominated by NET, even the canonically "floristically simple" boreal forest at BONA is quite mixed, even though it is classified in the csv as just being needleleaf evergreen.

I think that since we are interested in understanding both forest dynamics (FATES) as well as land-surface interactions (CTSM and FATES-CTSM), we would not want to discount a whole 40% of the forest that is photosynthesizing (or not) year-round (i.e. needleleaf evergreen trees vs. deciduous broadleaf trees), and has quite a different albedo, among other differences. I think this would impact both the FATES-CTSM runs as well as the CTSM Big-Leaf runs. 

Since we do have access to the actual data on PFT composition at the NEON sites it seems quite pertinent that we use that data to drive our model runs (as planned in #1934).

ekluzek replied:
Awesome, thanks for the analysis that's helpful. The naive question I wonder about is what the PFT mix will be based on the 1 degree grid cell for each of these sites? Is it going to be about right or include PFTs that really shouldn't be there? The % cover for each is going to be wrong for sure, but is it going to include PFTs that really aren't at the local site? I was thinking that would be the likely problem, but maybe PFTs are broad enough classes that you won't have that problem....

adrifoster replied:
I think that the PFTs are mostly broad enough that it won't make a huge difference.

For example, for the 1-degree surrounding HARV:

```
 PCT_NAT_PFT =
  0,
  15.5359300255115,
  0,
  0,
  0,
  0,
  0,
  78.9754444442113,
  0,
  0,
  0,
  0,
  0,
  5.47422440732277,
  0.014401122954473 ;

```
But we should try to update it using the data ASAP.

jkshuman replied:
Thanks @ekluzek We were talking about different datasets, and now that we all have the full information we can make an informed decision.

ekluzek replied:
Resolving this

adrifoster commented on tools/site_and_regional/subset_data:
```diff
@@ -4,22 +4,18 @@ This is a just top-level skeleton script that calls
 subset_data.py.
 The original code (subset_data.py) is located under
 python/ctsm folder.
-
```
sorry I think my editor did this without asking me.... I can put it back!

adrifoster commented on tools/site_and_regional/neon_surf_wrapper.py:
```diff
@@ -33,7 +33,7 @@
 
 import pandas as pd
-#import tqdm as tqdm
+
```
I am using the PyCharm IDE (by JetBrains), and I think it automatically deletes whitespace in some instances. Let me look into this and stop that from occurring automatically

ekluzek replied:
You should show us PyCharm sometime. I remember Sunniva recommended it, as does my daughter. I haven't used it though, and it helps to have someone demonstrate it a few times before trying it out.

adrifoster commented on tools/site_and_regional/modify_singlept_site_neon.py:
```diff
@@ -520,9 +535,9 @@ def main():
-    pdvers = pd.__version__
-    if version.parse(pdvers) < version.parse("1.1.0"):
-        sys.exit("The pandas version in your python environment is too old, update to a newer version of pandas (>=1.1.0): version=%s", pdvers )
+    #pdvers = pd.__version__
```
Okay, the problem here was that the new `ctsm_py` environment didn't have the `packages` python library installed, so we just need to add that.

adrifoster commented on cime_config/usermods_dirs/NEON/FATES/defaults/user_nl_clm:
```diff
@@ -0,0 +1,28 @@
+fsurdat = "$DIN_LOC_ROOT/lnd/clm2/surfdata_map/NEON/FATES/surfdata_1x1_NEON_16PFT_${NEONSITE}_hist_16pfts_Irrig_CMIP6_simyr2000_c230110.nc"
+
+! h1 output stream 
+hist_fincl2 = 'FATES_AUTORESP','FCEV','FCTR','FGEV','FIRA','FSA','FSH','FATES_GPP','FATES_GPP_PF','H2OSOI',
```
yeah it's annoying... unless we can put this part in the compset definition?

ekluzek commented:
We discuss this at the software meeting this morning. We are leaving it up to @adrifoster to either bring this in as is or implement my solution. In either case we will be looking at long term solutions to avoid duplication.

adrifoster commented on cime_config/usermods_dirs/NEON/FATES/ABBY/include_user_mods:
```diff
@@ -0,0 +1 @@
+../defaults
```
I think we are going to leave as is for now but with the understanding that we want to change this in the future to avoid duplication

adrifoster commented on cime_config/usermods_dirs/NEON/FATES/defaults/user_nl_clm:
```diff
@@ -0,0 +1,28 @@
+!----------------------------------------------------------------------------------
+
+flanduse_timeseries = ' '   ! This isn't needed for a non transient case, but will be once we start using transient compsets
+fsurdat = "$DIN_LOC_ROOT/lnd/clm2/surfdata_map/NEON/FATES/surfdata_1x1_NEON_16PFT_${NEONSITE}_hist_16pfts_Irrig_CMIP6_simyr2000_c230110.nc"
```
Just to check, since these surface files don't have a dominant PFT do we still want to change the directory name to 16pft? They are different from the CLM NEON versions in that they 1) have 16 pfts instead of 78 and 2) do not pick one dominant PFT for the gridcell

adrifoster replied:
Updating to note that we decided to change the flags and directories to `16pft`, `mixed`, and 16pft_mixed

jedwards4b commented:
I'm coming to this discussion late but it sounds like you have stuff in usermods_dirs/defaults that you want to apply only when its not a fates case - is that correct?   If so can you just change the include_user_mods in the non fates cases to
../defaults.nofates and in that directory have another include_user_mods pointing to defaults (which now contains only what you need for fates)

jedwards4b commented on cime_config/usermods_dirs/NEON/FATES/KONA/shell_commands:
```diff
@@ -0,0 +1,6 @@
+./xmlchange NEONSITE=KONA
+./xmlchange PTS_LON=263.38956
+./xmlchange PTS_LAT=39.10828
```
How about adding something like
./xmlchange BUILD_COMPLETE=FALSE

adrifoster replied:
Unfortunately while this works great for a cloned case, since shell_commands is usually called before the case.build command (?) we can still build and submit the case...

@ekluzek perhaps in this instance we should just remove the two crop user mods on the FATES side? We can add a README to the FATES directory to explain why these two are missing, and then when we add FATES to `./run_neon` we can easily add an error there for running those two sites until they are ready.

jedwards4b replied:
@adrifoster I think that your proposed solution makes sense.

ekluzek replied:
@adrifoster that makes sense to me as well. Thanks for trying the solution of having it error out -- it was worth doing. It's good to know how it functions in actual cases.

ekluzek replied:
We couldn't come up with a good way to handle this. So I'm resolving this conversation.

jkshuman reviewed (approved):
This looks great @adrifoster Thank you for bringing all this together and documenting it all so clearly.

adrifoster commented:
Testing on izumi and cheyenne finished (with some new expected fails):

Cheyenne:
`aux_clm` and `fates` had 1 new SHAREDLIB_BUILD fail for `ERP_P144x2_Ld30.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-mimicsFatesCold` due to the check I added in this PR for not allowing `-fire_emis` and `use_fates` (see #1045). I added this test to the expected fails until we fix this issue.

Izumi:
`aux_clm` had a few fails that were expected due to this PR: The new NEON-FATES-NIWO test was created in this PR so doesn't have a baseline to compare to. And the clm-NEON-NIWO uses a new surface dataset created in @wwieder's PR #1933
```
    FAIL SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.izumi_nag.clm-default--clm-NEON-NIWO NLCOMP
    FAIL SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.izumi_nag.clm-default--clm-NEON-NIWO BASELINE ctsm5.1.dev116: DIFF
    FAIL SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Fates.izumi_nag.clm-Fates--clm-NEON-FATES-NIWO NLCOMP
    FAIL SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Fates.izumi_nag.clm-Fates--clm-NEON-FATES-NIWO BASELINE ctsm5.1.dev116: ERROR BFAIL baseline directory '/fs/cgd/csm/ccsm_baselines/ctsm5.1.dev116/SMS_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Fates.izumi_nag.clm-Fates--clm-NEON-FATES-NIWO' does not exist
```

`fates` had the same failure for the baselines for the new fates NEON-NIWO test and then also had a timeout RUN failure for `ERS_D_Ld30.f45_f45_mg37.I2000Clm50FatesCruRsGs.izumi_nag.clm-FatesColdPRT2`. @glemieux and I tried to figure this one out but in the interest of getting this PR in I added to the expected fails list for now and created a [fates-side issue](https://github.com/NGEET/fates/issues/983) for it.

If we are okay with these test outcomes I think this PR is ready to go. @ekluzek?

ekluzek commented on tools/site_and_regional/subset_data:
```diff
@@ -9,17 +9,14 @@ For full instructions on how to run the code and different options,
 please check python/ctsm/subset_data.py file.
 This script extracts domain files, surface dataset, and DATM files
 at either a single point or a region using the global dataset.
-
```
@adrifoster could you restore this? Not a big deal...

adrifoster replied:
I thought I added that back! Not sure why it keeps disappearing...

ekluzek reviewed (approved):
I do have a couple small suggestions. But putting approve on this so as to not slow you down. This is a very nice feature addition that is so good to have in place.

We do have the caveat that we duplicated the NEON user-mods for FATES and we do want to change how that's done for maintainability. You should create an issue for that.

This is something that's my fault, but I made it so that 1x1_brazil is just 78PFT in ctsm5.1.dev116. Both mine and your testing seems to indicate that this is OK provided the simpler modes aren't used. I'm actually not sure why this even works as I thought it wouldn't. I'll create an issue on that one as it's my fault. I created issue #1948 for this one.

- [x] Create an issue about the duplication issue

ekluzek commented:
@adrifoster you should update the top description, since the argument names changed. It's useful to look back at closed PR's, and I usually just look at the description. So it is nice to have the description be accurate. That would help future me...

adrifoster commented:
> We do have the caveat that we duplicated the NEON user-mods for FATES and we do want to change how that's done for maintainability. You should create an issue for that.

Issue created here: #1949
